### DISCUSSION TITLE
Repeat from csv input data

### DISCUSSION THREAD

**Discussion by arnauramos1** [bonsai-rx/bonsai#1604] [Q&A] [answered]
Hi everyone,

I'm new to bonsai and I'm stuck with a simple task for which I have not found any related solution.. Maybe someone can help. 

The idea is simple:
 1) Read timepoints from a csv file.
 2) Trigger a certain stimmulus after each different timepoint.

The problem is that whenever the workflow repeats, it only gets the first value and does not continue on the following rows/ timepoints.

Thank you :) 

[readcsvtimes.zip](https://github.com/bonsai-rx/bonsai/files/13219159/readcsvtimes.zip)
![Screenshot 2023-10-31 175119](https://github.com/bonsai-rx/bonsai/assets/71004571/5588bf0a-b14f-42ea-a3ce-34fc07c653fa)

**Comment by chiyu1203** [comment]
My gut feeling is to wrap your modules (from Value to Speaker) into a CreateObserable and before Repeat, add Concat module. This kind of design works with my experiment but I do not know how to explain it.

**Reply by bruno-f-cruz** [reply, marked as answer]
That is correct. You can only change the DueTime property of the Timer prior to the Subscription. In your case it will just use the first value that CsvReader reads. Since you are repeating the whole sequence, you are effectively reloading the csv file each time you repeat, so you will just draw the first value over and over again. Here's a potential solution:

![image](https://github.com/bonsai-rx/bonsai/assets/7049351/e0673d07-bca9-4b94-81d8-19338b083b7b)

[readcsvtimes.zip](https://github.com/bonsai-rx/bonsai/files/13228417/readcsvtimes.zip)

**Reply by arnauramos1** [reply]
Thank you both very much!
